bugalink-backend/trips/views.py

```diff
@@ -327,3 +330,100 @@ def get(self, request, *args, **kwargs):
         return Response(
             TripSerializer(trips, many=True).data, status=status.HTTP_200_OK
         )
+
+
+class TripRateViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = TripRequest.objects.all()
+    serializer_class = TripRequestSerializer
+
+    @action(detail=True, methods=["post"])
+    def post(self, request, trip_id, *args, **kwargs):
+        try:
+            trip = Trip.objects.get(id=trip_id)
+            if trip.status == "FINISHED":
+                trip_request = TripRequest.objects.filter(
+                    trip=trip, status="ACCEPTED", passenger__user=request.user
+                ).first()
+            if trip_request:
+                rating = request.POST.get("rating")
+                is_good_driver = request.POST.get("is_good_driver")
+                is_pleasant_driver = request.POST.get("is_pleasant_driver")
+                already_knew = request.POST.get("already_knew")
+
+                DriverRating.objects.create(
+                    trip_request=TripRequest.objects.get(id=trip_request.id),
+                    rating=rating,
+                    is_good_driver=is_good_driver,
+                    is_pleasant_driver=is_pleasant_driver,
+                    already_knew=already_knew,
+                )
+
+                return Response({"message": "Valoración realizada con exito"})
+            else:
+                return Response({"message": "No has participado en este viaje"})
+        except Exception as e:
+            return Response({"message": str(e)})
+
+
+class ReportIssueViewSet(
+    mixins.RetrieveModelMixin,
+    mixins.DestroyModelMixin,
+    viewsets.GenericViewSet,
+):
+    queryset = Trip.objects.all()
+    serializer_class = TripSerializer
+
+    @action(detail=True, methods=["post"])
+    def post(self, request, trip_id, *args, **kwargs):
+        try:
+            trip = Trip.objects.get(id=trip_id, status="FINISHED")
+            user = request.user
+            trip_request = TripRequest.objects.filter(
+                trip=trip, passenger__user=user
+            ).first()
+            if trip_request or trip.driver_routine.driver.user == user:
+                reported_user_id = request.POST.get("reported_user_id")
+                reported_user = User.objects.get(id=reported_user_id)
+                if user == trip.driver_routine.driver.user:
+                    reporter_is_driver = True
+                    reported_is_driver = False
+                elif reported_user == trip.driver_routine.driver.user:
+                    reported_is_driver = True
+                    reporter_is_driver = False
+                else:
+                    reported_is_driver = False
```

Comment:
A much cleaner way to have this code, instead of 10 lines with 3 ifs, is the following:

```python
reporter_is_driver = user == trip.driver_routine.driver.user
reported_is_driver = reported_user == trip.driver_routine.driver.user
```